internal/host/plugin/host_catalog.go

```diff
@@ -35,16 +35,22 @@ func NewHostCatalog(ctx context.Context, scopeId, pluginId string, opt ...Option
 			Name:        opts.withName,
 			Description: opts.withDescription,
 		},
-		secrets: opts.withSecrets,
 	}
 
 	if opts.withAttributes != nil {
 		attrs, err := json.Marshal(opts.withAttributes)
 		if err != nil {
-			return nil, errors.Wrap(ctx, err, op, errors.WithCode(errors.InvalidParameter))
+			return nil, errors.Wrap(ctx, err, op, errors.WithCode(errors.InvalidParameter), errors.WithMsg("marshaling attributes"))
 		}
 		hc.Attributes = attrs
 	}
+	if opts.withSecrets != nil {
+		secrets, err := json.Marshal(opts.withSecrets)
```

Comment:
Given that secrets comes in as a structpb.Struct and is passed to plugins as a structpb.Struct, why are we using JSON to marshal it? Seems like we should just proto.Marshal the struct.

Reply:
I have fixed this.  It was an experiment I was playing with in this PR but never reverted.  Thanks for catching it.